internal/resources/private_endpoints.go

```diff
@@ -0,0 +1,317 @@
+		return
+	}
+
+	refreshedState, err := p.getPrivateEndpointState(ctx, organizationId, projectId, clusterId, endpointId)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error reading private endpoint service status",
+			"Error reading private endpoint service status, unexpected error: "+err.Error(),
+		)
+
+		return
+	}
+
+	diags = resp.State.Set(ctx, refreshedState)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func (p *PrivateEndpoint) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state providerschema.PrivateEndpoint
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	IDs, err := state.Validate()
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error Reading Private Endpoint",
+			"Could not validate private endpoint "+state.EndpointId.String()+": "+err.Error(),
+		)
+		return
+	}
+
+	var (
+		organizationId = IDs[providerschema.OrganizationId]
+		projectId      = IDs[providerschema.ProjectId]
+		clusterId      = IDs[providerschema.ClusterId]
+		endpointId     = IDs[providerschema.EndpointId]
+	)
+
+	refreshedState, err := p.getPrivateEndpointState(ctx, organizationId, projectId, clusterId, endpointId)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error reading private endpoint status",
+			"Error reading private endpoint status, unexpected error: "+err.Error(),
+		)
+
+		return
+	}
+
+	diags = resp.State.Set(ctx, &refreshedState)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Update there is no update API so returns an error.
+func (p *PrivateEndpoint) Update(_ context.Context, _ resource.UpdateRequest, resp *resource.UpdateResponse) {
+	resp.Diagnostics.AddError(
+		"No update API for a private endpoint",
+		"No update API for a private endpoint",
+	)
+}
+
+// Delete rejects a private endpoint on the CSP.
+func (p *PrivateEndpoint) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var state providerschema.PrivateEndpoint
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	IDs, err := state.Validate()
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error rejecting private endpoint",
+			"Could not reject endpoint due to validation error: "+err.Error(),
+		)
+		return
+	}
+
+	var (
+		organizationId = IDs[providerschema.OrganizationId]
+		projectId      = IDs[providerschema.ProjectId]
+		clusterId      = IDs[providerschema.ClusterId]
+		endpointId     = IDs[providerschema.EndpointId]
+	)
+
+	url := fmt.Sprintf(
+		"%s/v4/organizations/%s/projects/%s/clusters/%s/privateEndpointService/endpoints/%s/unassociate",
+		p.HostURL,
+		organizationId,
+		projectId,
+		clusterId,
+		endpointId,
+	)
+	cfg := api.EndpointCfg{Url: url, Method: http.MethodPost, SuccessStatus: http.StatusNoContent}
```

Comment:
Why is this `MethodPost`? 
```suggestion
	cfg := api.EndpointCfg{Url: url, Method: http.MethodDelete, SuccessStatus: http.StatusNoContent}
```

Reply:
+1

Reply:
an endpoint is deleted/rejected with `POST /unassociate` .  conversely, to add/accept endpoint we use `POST /associate`.  this is modeled after CMEK.